source/draw/gpu/opengl/Map/OpenGLMapPack.ooc

```diff
@@ -81,6 +84,42 @@ OpenGLMapPackUv: class extends OpenGLMap {
 		}
 		"
 }
+OpenGLMapPackUvPadded: class extends OpenGLMap {
+	init: func (context: OpenGLContext) { super(This vertexSource, This fragmentSource, context) }
+	vertexSource: static String ="#version 300 es
+		precision mediump float;
+		uniform mat4 transform;
+		uniform float xOffset;
+		uniform float paddingOffset;
+		uniform float texelOffset;
+		layout(location = 0) in vec2 vertexPosition;
+		layout(location = 1) in vec2 textureCoordinate;
+		out vec2 fragmentTextureCoordinate[2];
+		void main() {
+			fragmentTextureCoordinate[0] = textureCoordinate + vec2(-xOffset - paddingOffset, 0);
+			fragmentTextureCoordinate[1] = textureCoordinate + vec2(texelOffset - xOffset - paddingOffset, 0);
+			gl_Position = transform * vec4(vertexPosition.x, vertexPosition.y, -1, 1);
+		}"
+	fragmentSource: static String ="#version 300 es
+		precision mediump float;
+		uniform sampler2D texture0;
+		uniform float rowUnit;
+		in highp vec2 fragmentTextureCoordinate[2];
+		out vec4 outColor;
+		void main() {
+			vec2 shiftedCoor = vec2(fragmentTextureCoordinate[0].x, fragmentTextureCoordinate[0].y);
+			if (shiftedCoor.x < 0.0) {
+				shiftedCoor = vec2(1.0 + shiftedCoor.x, shiftedCoor.y - rowUnit);
+			}
+			vec2 rg = texture(texture0, shiftedCoor).rg;
+			shiftedCoor = vec2(fragmentTextureCoordinate[1].x, fragmentTextureCoordinate[1].y);
+			if (shiftedCoor.x < 0.0) {
+				shiftedCoor = vec2(1.0 + shiftedCoor.x, shiftedCoor.y - rowUnit);
+			}
+			vec2 ba = texture(texture0, shiftedCoor).rg;
+			outColor = vec4(rg.x, rg.y, ba.x, ba.y);
+		}"
+}
```

Comment:
No but I will make big PR later where I structure all shaders into files and `slurp` so I didn't do that here.